### PR TITLE
[Merged by Bors] - chore: remove >1 year deprecations

### PR DESCRIPTION
**Pull request by kim-em**
This isn't everything with >1year deprecation ages: unfortunately lots is still actually in use!

### CODE REVIEW

**Comment by vihdzp** [comment]
It'd be nice to have a list of the old deprecated theorems in current use, so people can get to cleaning them up.

**Comment by semorrison** [comment]
> It'd be nice to have a list of the old deprecated theorems in current use, so people can get to cleaning them up.

The answer here is basically: "any lemma with `nthLe` in the name".

**Comment by semorrison** [comment]
@Parcly-Taxel, thanks for taking a look here. It would actually make more sense if you made that commit into a separate PR:
1. I could then immediately merge it, instead of us needing to go find another reviewer
2. It will keep this PR simpler, as it is just deleting things, not making significant changes.

**Comment by Parcly-Taxel** [comment]
> @Parcly-Taxel, thanks for taking a look here. It would actually make more sense if you made that commit into a separate PR:
> 
>     1. I could then immediately merge it, instead of us needing to go find another reviewer
> 
>     2. It will keep this PR simpler, as it is just deleting things, not making significant changes.

#15965

File: Mathlib/Data/List/Cycle.lean [hunk 1]
```diff
@@ -273,106 +274,93 @@ theorem next_get : ∀ (l : List α) (_h : Nodup l) (i : Fin l.length),
...
-        refine h _ _ hn Nat.succ_pos' ?_
-        simpa using H
-      · intro H
-        suffices n.succ.succ = 1 by simpa
-        rw [nodup_iff_nthLe_inj] at h
-        refine h _ _ hn (Nat.succ_lt_succ Nat.succ_pos') ?_
-        simpa using H
-
-set_option linter.deprecated false in
+theorem prev_get (l : List α) (_h : Nodup l) (i : Fin l.length) :
```
Comment (mattrobball) [inline Mathlib/Data/List/Cycle.lean:277]: I would prefer that we explicitly turn off the linter and comment about how it is failing at `h` than work around the issues.

Comment (semorrison) [reply Mathlib/Data/List/Cycle.lean:277]: @Parcly-Taxel pushed these changes to my PR ... :-(

Comment (semorrison) [reply Mathlib/Data/List/Cycle.lean]: I've deleted Jeremy's commit from this PR, so this comment now belongs on https://github.com/leanprover-community/mathlib4/pull/15965.

Comment (semorrison) [reply Mathlib/Data/List/Cycle.lean]: (And I've implemented the suggestion there.)

**Comment by grunweg** [comment]
Thanks!
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**